scalafix-testkit/src/main/scala/scalafix/testkit/RuleTest.scala

```diff
@@ -31,36 +30,48 @@ object RuleTest {
       val input = test.toInput
       val dialect = args.scalaVersion.dialect(args.sourceScalaVersion)
       val tree = dialect(input).parse[Source].get
-      val comment = SemanticRuleSuite.findTestkitComment(tree.tokens)
-      val syntax = comment.syntax.stripPrefix("/*").stripSuffix("*/")
-      val conf = Conf.parseString(test.testName, syntax).get
-      val scalafixConfig = conf.as[ScalafixConfig].get
-      val doc = v1.SyntacticDocument(
-        tree.pos.input,
-        LazyValue.now(tree),
-        DiffDisable.empty,
-        scalafixConfig
-      )
-      val sdoc =
-        v1.SemanticDocument.fromPath(
-          doc,
-          test.semanticdbPath,
-          classLoader,
-          symtab
-        )
-      val decoderSettings =
-        RuleDecoder.Settings().withConfig(scalafixConfig)
-      val decoder = RuleDecoder.decoder(decoderSettings)
-      val rulesConf = ConfGet
-        .getKey(conf, "rules" :: "rule" :: Nil)
-        .getOrElse(Conf.Lst(Nil))
-      val config = Configuration()
-        .withConf(conf)
-        .withScalaVersion(args.scalaVersion.value)
-        .withScalacOptions(args.scalacOptions)
-        .withScalacClasspath(args.classpath.entries)
-      val rules = decoder.read(rulesConf).get.withConfiguration(config).get
-      (rules, sdoc)
+      val allComments = SemanticRuleSuite.filterPossibleTestkitComments(tree.tokens)
+      val lazilyParsedComments = allComments.view.map{ comment =>
+        val syntax = comment.syntax.stripPrefix("/*").stripSuffix("*/")
+        Conf.parseString(test.testName, syntax) match {
+          case Configured.Ok(conf) =>
+            val scalafixConfig = conf.as[ScalafixConfig].get
+            val doc = v1.SyntacticDocument(
+              tree.pos.input,
+              LazyValue.now(tree),
+              DiffDisable.empty,
+              scalafixConfig
+            )
+            val sdoc =
+              v1.SemanticDocument.fromPath(
+                doc,
+                test.semanticdbPath,
+                classLoader,
+                symtab
+              )
+            val decoderSettings =
+              RuleDecoder.Settings().withConfig(scalafixConfig)
+            Some((RuleDecoder.decoder(decoderSettings), conf, sdoc))
+          case Configured.NotOk(_) => None
+        }
+      }
+
+      lazilyParsedComments.collectFirst {
+        case Some((decoder, conf, sdoc)) =>
+          val rulesConf = ConfGet
+            .getKey(conf, "rules" :: "rule" :: Nil)
+            .getOrElse(Conf.Lst(Nil))
```

Comment:
per the message in the exception, this should be lifted up to the `lazilyParsedComments` closure to effectively skip comments that are HOCON but do not contain `rule` or `rules`

Reply:
So I thought the failure in parsing is done in `Conf.parseString(test.testName, syntax)` (which is why I pattern match against it). The idea behind my solution is that we create a lazy collection called `lazilyParsedComments` of successfully/unsuccessfully parsed comments and then we do `collectFirst` to get the first successful parsing.

In other words, the logic of figuring out what is a correct HOCON comment has already happened by the time we get to `rulesConf`

Reply:
> This is definitely an improvement and should cover your original use-case of a long comment

Actually, it does not - the test I added in fd35cf2b6e63fb251e73a152a0badede1b93ab75 shows it. I will try to do what I was suggesting in another commit and see if it helps.